### PR TITLE
Partial fix for #143; TUI Improvements; Resolve #84; #106; #157

### LINKED ISSUE #84: Add Switch To Testing Branch
Once #46 is fixed by #83, we should add some kind of flag or option to switch to our https://github.com/FaithLife-Community/test-builds repo, so that we can have a Stable and a Testing/Rolling-Release version of the app. This would make testing new changes a breeze.

### LINKED ISSUE #106: Winetricks Quick Action Buttons
While we have a button to launch winetricks and a couple others, there are some common debugging kinds of things we could implement as quick action buttons, or at least as a submenu for the TUI.

Here I am thinking of a button to change the graphics renderer, given some of the compositing issues we face. Navigating the winetricks menu is not the most user friendly of things. And while I think there is value in making others learn how to use new tools, some things are common to our program and I think easily fall under the idea of using our program for maintenance tools.

I might use https://github.com/RPi-Distro/raspi-config as inspiration.

cf. b588e9b84f181ced0b17b8f1ae1eb62f0d2532d7 for a partial implementation.

Besides those above, others to consider:

- Install Fonts
- Change Renderer
- Change Windows OS version

### PR DESCRIPTION
**Pull request by thw26**
This PR helps to fix at least one of the issues in #143.

I've also swapped the choice processor to using a dict to make it easier to understand or even reuse code, as per #146.

I've also added in the ability to switch the TUI's color scheme. See bottom of main menu for button.

I've also fixed its vertical and horizontal resizing and made it so the console log grows in size, too.

This PR also adds in submenus to the TUI to reduce the number of options in any single menu.

Implements a config and a TUI solution to fix #84 and #157.

This PR resolves #106 for the TUI.

This PR also adds in logging deduplication, to avoid unnecessary additional logs.

### CODE REVIEW

**Comment by thw26** [comment]
After implementing the Winetricks options, for some reason the winetricks commands are not working. We need to figure out why.

File: installer.py [hunk 1]
```diff
@@ -559,6 +559,54 @@ def ensure_icu_data_files(app=None):
     logging.debug('> ICU data files installed')
 
 
+def ensure_winetricks_applied(app=None):
```
Comment (thw26) [inline installer.py:562]: I'm not sure how this was deleted

**Comment by thw26** [comment]
There is some kind of error when running winetricks in the TUI that causes a winetricks command to be run twice:

```
2024-09-18 23:40:39 INFO: running "winetricks -q settings renderer=gdi"
2024-09-18 23:40:39 DEBUG: Getting wine environment.
2024-09-18 23:40:39 DEBUG: dynamic_path: /home/thwright/LogosBible10/data/bin/wine-devel_9.12-x86_64.AppImage
2024-09-18 23:40:39 DEBUG: wine_exe_path=PosixPath('/home/thwright/LogosBible10/data/bin/wine-devel_9.12-x86_64.AppImage')
2024-09-18 23:40:39 DEBUG: Wine env: {'WINE': '/home/thwright/LogosBible10/data/bin/wine64', 'WINEDEBUG': 'fixme-all,err-all', 'WINEDLLOVERRIDES': '', 'WINELOADER': '/home/thwright/LogosBible10/data/bin/wine64', 'WINEPREFIX': '/home/thwright/LogosBible10/data/wine64_bottle', 'WINESERVER': '/home/thwright/LogosBible10/data/bin/wineserver'}
2024-09-18 23:40:39 DEBUG: run_wine_proc: /usr/bin/winetricks; exe=None; exe_args=['-q', 'settings', 'renderer=gdi']
2024-09-18 23:40:39 DEBUG: subprocess cmd: '/usr/bin/winetricks -q settings renderer=gdi'
2024-09-18 23:40:44 DEBUG: ------------------------------------------------------
2024-09-18 23:40:44 DEBUG: warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
2024-09-18 23:40:44 DEBUG: ------------------------------------------------------
2024-09-18 23:40:44 DEBUG: warning: You apppear to be using Wine's new wow64 mode. Note that this is EXPERIMENTAL and not yet fully supported. If reporting an issue, be sure to mention this.
2024-09-18 23:40:44 DEBUG: ------------------------------------------------------
2024-09-18 23:40:44 DEBUG: Using winetricks 20240105 - sha256sum: 8f28985b06e9e04b5268e2aef3429b2a0f6abd4fd12c4a30472dfe66355128fa with wine-9.12 and WINEARCH=win64
2024-09-18 23:40:45 DEBUG: Executing w_do_call renderer=gdi
2024-09-18 23:40:49 DEBUG: ------------------------------------------------------
2024-09-18 23:40:49 DEBUG: warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
2024-09-18 23:40:49 DEBUG: ------------------------------------------------------
2024-09-18 23:40:49 DEBUG: warning: You apppear to be using Wine's new wow64 mode. Note that this is EXPERIMENTAL and not yet fully supported. If reporting an issue, be sure to mention this.
2024-09-18 23:40:49 DEBUG: ------------------------------------------------------
2024-09-18 23:40:49 DEBUG: Executing load_renderer gdi
2024-09-18 23:40:49 DEBUG: Setting Direct3D/renderer to gdi
2024-09-18 23:40:49 DEBUG: Executing /home/thwright/LogosBible10/data/bin/wine64 C:\windows\syswow64\regedit.exe /S C:\windows\Temp\set-wined3d.reg
2024-09-18 23:40:50 DEBUG: Executing /home/thwright/LogosBible10/data/bin/wine64 C:\windows\regedit.exe /S C:\windows\Temp\set-wined3d.reg
2024-09-18 23:40:50 INFO: "winetricks -q settings renderer=gdi" DONE!
2024-09-18 23:40:50 INFO: * Starting wait_process_using_dir for /home/thwright/LogosBible10/data/wine64_bottle…
2024-09-18 23:40:50 INFO: * End of wait_process_using_dir.
2024-09-18 23:40:50 DEBUG: dynamic_path: /home/thwright/LogosBible10/data/bin/wine-devel_9.12-x86_64.AppImage
2024-09-18 23:40:50 DEBUG: wine_exe_path=PosixPath('/home/thwright/LogosBible10/data/bin/wine-devel_9.12-x86_64.AppImage')
2024-09-18 23:40:50 DEBUG: Wine env: {'WINE': '/home/thwright/LogosBible10/data/bin/wine64', 'WINEDEBUG': 'fixme-all,err-all', 'WINEDLLOVERRIDES': '', 'WINELOADER': '/home/thwright/LogosBible10/data/bin/wine64', 'WINEPREFIX': '/home/thwright/LogosBible10/data/wine64_bottle', 'WINESERVER': '/home/thwright/LogosBible10/data/bin/wineserver'}
2024-09-18 23:40:53 INFO: "/home/thwright/LogosBible10/data/bin/wineserver -w" has ended properly.
2024-09-18 23:40:53 INFO: running "winetricks -q settings renderer=Return to Main Menu"
2024-09-18 23:40:53 DEBUG: Getting wine environment.
2024-09-18 23:40:53 DEBUG: dynamic_path: /home/thwright/LogosBible10/data/bin/wine-devel_9.12-x86_64.AppImage
2024-09-18 23:40:53 DEBUG: wine_exe_path=PosixPath('/home/thwright/LogosBible10/data/bin/wine-devel_9.12-x86_64.AppImage')
2024-09-18 23:40:53 DEBUG: Wine env: {'WINE': '/home/thwright/LogosBible10/data/bin/wine64', 'WINEDEBUG': 'fixme-all,err-all', 'WINEDLLOVERRIDES': '', 'WINELOADER': '/home/thwright/LogosBible10/data/bin/wine64', 'WINEPREFIX': '/home/thwright/LogosBible10/data/wine64_bottle', 'WINESERVER': '/home/thwright/LogosBible10/data/bin/wineserver'}
2024-09-18 23:40:53 DEBUG: run_wine_proc: /usr/bin/winetricks; exe=None; exe_args=['-q', 'settings', 'renderer=Return to Main Menu']
2024-09-18 23:40:53 DEBUG: subprocess cmd: '/usr/bin/winetricks -q settings renderer=Return to Main Menu'
2024-09-18 23:40:58 DEBUG: ------------------------------------------------------
2024-09-18 23:40:58 DEBUG: warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
2024-09-18 23:40:58 DEBUG: ------------------------------------------------------
2024-09-18 23:40:58 DEBUG: warning: You apppear to be using Wine's new wow64 mode. Note that this is EXPERIMENTAL and not yet fully supported. If reporting an issue, be sure to mention this.
2024-09-18 23:40:58 DEBUG: ------------------------------------------------------
2024-09-18 23:40:59 DEBUG: Using winetricks 20240105 - sha256sum: 8f28985b06e9e04b5268e2aef3429b2a0f6abd4fd12c4a30472dfe66355128fa with wine-9.12 and WINEARCH=win64
2024-09-18 23:41:00 DEBUG: Unknown arg renderer=Return to Main Menu
2024-09-18 23:41:00 DEBUG: Usage: /usr/bin/winetricks [options] [command|verb|path-to-verb] ...
2024-09-18 23:41:00 DEBUG: Executes given verbs.  Each verb installs an application or changes a setting.
2024-09-18 23:41:00 DEBUG: 
2024-09-18 23:41:00 DEBUG: Options:
2024-09-18 23:41:00 DEBUG:     --country=CC      Set country code to CC and don't detect your IP address
2024-09-18 23:41:00 DEBUG: -f, --force           Don't check whether packages were already installed
2024-09-18 23:41:00 DEBUG:     --gui             Show gui diagnostics even when driven by commandline
2024-09-18 23:41:00 DEBUG:     --gui=OPT         Set OPT to kdialog or zenity to override GUI engine
2024-09-18 23:41:00 DEBUG:     --isolate         Install each app or game in its own bottle (WINEPREFIX)
2024-09-18 23:41:00 DEBUG:     --self-update     Update this application to the last version
2024-09-18 23:41:00 DEBUG:     --update-rollback Rollback the last self update
2024-09-18 23:41:00 DEBUG: -k, --keep_isos       Cache isos (allows later installation without disc)
2024-09-18 23:41:00 DEBUG:     --no-clean        Don't delete temp directories (useful during debugging)
2024-09-18 23:41:00 DEBUG:     --optin           Opt in to reporting which verbs you use to the Winetricks maintainers
2024-09-18 23:41:00 DEBUG:     --optout          Opt out of reporting which verbs you use to the Winetricks maintainers
2024-09-18 23:41:00 DEBUG: -q, --unattended      Don't ask any questions, just install automatically
2024-09-18 23:41:00 DEBUG: -r, --ddrescue        Retry hard when caching scratched discs
2024-09-18 23:41:00 DEBUG: -t  --torify          Run downloads under torify, if available
2024-09-18 23:41:00 DEBUG:     --verify          Run (automated) GUI tests for verbs, if available
2024-09-18 23:41:00 DEBUG: -v, --verbose         Echo all commands as they are executed
2024-09-18 23:41:00 DEBUG: -h, --help            Display this message and exit
2024-09-18 23:41:00 DEBUG: -V, --version         Display version and exit
2024-09-18 23:41:00 DEBUG: 
2024-09-18 23:41:00 DEBUG: Commands:
2024-09-18 23:41:00 DEBUG: list                  list categories
2024-09-18 23:41:00 DEBUG: list-all              list all categories and their verbs
2024-09-18 23:41:00 DEBUG: apps list             list verbs in category 'applications'
2024-09-18 23:41:00 DEBUG: benchmarks list       list verbs in category 'benchmarks'
2024-09-18 23:41:00 DEBUG: dlls list             list verbs in category 'dlls'
2024-09-18 23:41:00 DEBUG: fonts list            list verbs in category 'fonts'
2024-09-18 23:41:00 DEBUG: settings list         list verbs in category 'settings'
2024-09-18 23:41:00 DEBUG: list-cached           list cached-and-ready-to-install verbs
2024-09-18 23:41:00 DEBUG: list-download         list verbs which download automatically
2024-09-18 23:41:00 DEBUG: list-manual-download  list verbs which download with some help from the user
2024-09-18 23:41:00 DEBUG: list-installed        list already-installed verbs
2024-09-18 23:41:00 DEBUG: arch=32|64            create wineprefix with 32 or 64 bit, this option must be
2024-09-18 23:41:00 DEBUG:                       given before prefix=foobar and will not work in case of
2024-09-18 23:41:00 DEBUG:                       the default wineprefix.
2024-09-18 23:41:00 DEBUG: prefix=foobar         select WINEPREFIX=/home/thwright/.local/share/wineprefixes/foobar
2024-09-18 23:41:00 DEBUG: annihilate            Delete ALL DATA AND APPLICATIONS INSIDE THIS WINEPREFIX
2024-09-18 23:41:00 ERROR: Error running '/usr/bin/winetricks -q settings renderer=Return to Main Menu': 1
2024-09-18 23:41:00 CRITICAL: "['/usr/bin/winetricks', '-q', 'settings', 'renderer=Return to Main Menu']" failed with exit code: 1
```

**Comment by n8marti** [comment]
Also looks good when testing the binary in Ubuntu 20.04 and Debian 12. There are problems with dependencies in Arch, but I don't think that needs to be dealt with in this PR

**Review by n8marti — APPROVED** [review]

**Pull request merged**